src/Raven.Client/Documents/Subscriptions/SubscriptionCriteria.cs

```diff
@@ -124,19 +151,54 @@ public override void ConvertToJavascript(JavascriptConversionContext context)
 
                 using (javascriptWriter.Operation(node))
                 {
-                    if (node.Expression == Parameter)
+                    if (node.Type == typeof(DateTime))
```

Comment:
Please add comments here showing the C# you are matching and the JS that is being generated.